app/screens/settings/settings.tsx

```diff
@@ -93,33 +93,9 @@ const Settings = ({componentId, showHelp, siteName}: SettingsProps) => {
         });
     }, []);
 
-    useEffect(() => {
-        const backHandler = BackHandler.addEventListener('hardwareBackPress', () => {
-            if (NavigationStore.getNavigationTopComponentId() === componentId) {
-                close();
-                return true;
-            }
-
-            return false;
-        });
-        return () => {
-            backHandler.remove();
-        };
-    }, []);
+    useAndroidHardwareBackHandler(componentId, close);
 
-    useEffect(() => {
-        const unsubscribe = Navigation.events().registerComponentListener({
-            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
-                if (buttonId === CLOSE_BUTTON_ID) {
-                    close();
-                }
-            },
-        }, componentId);
-
-        return () => {
-            unsubscribe.remove();
-        };
-    }, []);
+    useNavButtonPressed(CLOSE_BUTTON_ID, componentId, close, [close]);
```

Comment:
close does not need to be memoized, and it would not be needed as a dep

Reply:
corrected